p2p/proto/block.proto

```diff
@@ -0,0 +1,80 @@
+syntax = "proto3";
+import "p2p/proto/common.proto";
+import "p2p/proto/state.proto";
+import "google/protobuf/timestamp.proto";
+
+// for now, we assume a small consensus, so this fits in 1M. Else, these will be repeated
+message Signatures {
+    uint64 block_number = 1;
+
+    repeated ConsensusSignature signatures = 2; //
+                                       // can be more explicit here about the signature structure as this is not part of account abstraction
+}
+
+// Note: commitments may change to be for the previous blocks like comet/tendermint
+// hash of block header sent to L1
+message BlockHeader {
+    Hash                      parent_header     = 1;
+    uint64                    number            = 2;
+    google.protobuf.Timestamp time              = 3;   // TODO: see if this needs to be Felt252 or can be converted
+    Address                   sequencer_address = 4;
+    Merkle                    state_diffs       = 5;   //  By order of (contract, key), taking last in case of duplicates.
+                                                       // This means the proposer needs to sort after finishing the block (TBD: patricia? )
+                                                       // State is optional and appears every X blocks for the last block. This is to support
+                                                       // snapshot sync and also so that light nodes can sync on state without state diffs.
+    Patricia                  state             = 6;   // hash of contract and class patricia tries. Same as in L1. Later more trees will be included
+    Hash                      proof_fact        = 7;   // for Kth block behind. A hash of the output of the proof
+
+    // The following merkles can be built on the fly while sequencing/validating txs.
+    Merkle                    transactions      = 8;   // By order of execution. TBD: required? the client can execute (powerful machine) and match state diff
+    Merkle                    events            = 9;   // By order of issuance. TBD: in receipts?
+    Merkle                    receipts          = 10;  // By order of issuance.
+}
+
+message BlockProof {
+    bytes proof = 1; // proof size is currently 142K
+}
+
+// sent to all peers (except the ones this was received from, if any).
+// for a fraction of peers, also send the GetBlockHeaders and GetBlockBodies response (as if they asked for it for this block)
+message NewBlock {
+    oneof maybe_full {
+        uint64               block_number = 1;
+        BlockHeadersResponse header       = 2;
+        BlockBodiesResponse  body         = 3;
+    }
+}
+
+
+// result is (BlockHeader, Signature?)* in order of creation (incr/dec)
+message BlockHeadersRequest {
+    Iteration iteration = 1;
+}
+
+message BlockHeadersResponse {
+    uint64 block_number = 1;
```

Comment:
If fin is `error` should the `block_number` be the value at which we encountered the error?
What if fin is `ok`? Should then the block_number point to the last valid block/response in this stream?

Reply:
@ittaysw do you mind explaining why we need the `fin` again?
Each `BlockHeader` message delimits messages related to each block and when the peer is done responding, it will close the stream anyways?

Reply:
@CHr15F0x in both cases it is the block number. There can be several in a stream, per block. 
@omerfirmak As a future preparation for when we interleave responses

Reply:
The idea was 'fin' marks each block data, as it can span multiple messages. In case of an error overall, the fin can have a 0 block number